io.asgardio.java.oidc.sdk/src/main/java/io/asgardio/java/oidc/sdk/request/OIDCRequestBuilder.java

```diff
@@ -58,62 +63,83 @@ public OIDCRequestBuilder(OIDCAgentConfig oidcAgentConfig) {
-                .state(stateParameter)
-                .redirectionURI(callBackURI)
+        AuthenticationRequest authenticationRequest = new AuthenticationRequest.Builder(responseType, authScope,
+                clientID, callBackURI)
+                .state(state)
                 .endpointURI(authorizationEndpoint)
+                .nonce(nonce)
                 .build();
-        return authorizationRequest.toURI().toString();
+
+        io.asgardio.java.oidc.sdk.request.model.AuthenticationRequest authRequest =
+                new io.asgardio.java.oidc.sdk.request.model.AuthenticationRequest(authenticationRequest.toURI(),
+                        requestContext);
+
+        return authRequest;
     }
 
     /**
-     * Returns {@link String} Logout request. To build the logout request,
+     * Returns {@link io.asgardio.java.oidc.sdk.request.model.LogoutRequest} Logout request. To build the logout request,
      * {@link OIDCAgentConfig} should contain:
      * <ul>
      * <li>The logout endpoint URI
      * <li>The post logout redirection URI
      * </ul>
      *
-     * @param authenticationInfo {@link AuthenticationInfo} object with information of the current LoggedIn session.
-     *                           It must include a valid ID token.
-     * @param state              State parameter.
+     * @param sessionContext {@link SessionContext} object with information of the current LoggedIn session.
+     *                       It must include a valid ID token.
      * @return Logout request.
      */
-    public String buildLogoutRequest(AuthenticationInfo authenticationInfo, String state) {
+    public io.asgardio.java.oidc.sdk.request.model.LogoutRequest buildLogoutRequest(SessionContext sessionContext)
+            throws SSOAgentServerException {
 
         URI logoutEP = oidcAgentConfig.getLogoutEndpoint();
         URI redirectionURI = oidcAgentConfig.getPostLogoutRedirectURI();
-        JWT jwtIdToken = authenticationInfo.getIdToken();
-        State stateParam = null;
+        JWT jwtIdToken = null;
+        try {
+            jwtIdToken = JWTParser.parse(sessionContext.getIdToken());
+        } catch (ParseException e) {
+            throw new SSOAgentServerException(e.getMessage(), e);
+        }
+        State state = generateStateParameter();
+        RequestContext requestContext = new RequestContext();
 
-        if (StringUtils.isNotBlank(state)) {
-            stateParam = new State(state);
+        requestContext.setState(state);
+        URI logoutRequestURI;
+
+        try {
+            logoutRequestURI = new LogoutRequest(logoutEP, jwtIdToken, redirectionURI, state).toURI();
+        } catch (Exception e) {
+            throw new SSOAgentServerException(e.getMessage(), e);
         }
-        return new LogoutRequest(logoutEP, jwtIdToken, redirectionURI, stateParam).toURI().toString();
+
+        return new io.asgardio.java.oidc.sdk.request.model.LogoutRequest(logoutRequestURI, requestContext);
```

Comment:
Can we import the class without prepending the package name everywhere?

Reply:
We already have a import called com.nimbusds.openid.connect.sdk.LogoutRequest which refers to the LogoutRequest class in the nimbusds package. 
To distinguish between the two LogoutRequest classes, the package name had to be prepended.
Suggestions to resolve this are welcome.